modules/deb.sh

```diff
@@ -95,31 +96,67 @@ function deb_from_repo_pool_get()  {
     fi
 }
 
-# Extract the binary files of a deb package into a directory
+# Install a deb package using a chroot
 #
 #  $1 - Deb package
 #  $2 - Dest directory
 #
-function deb_extract_package()  {
+function deb_install_package() {
     if [[ $# -ne 2 ]]; then
-        log_fatal "deb_extract_package() requires 2 arguments"
+        log_fatal "deb_install_package() requires 2 arguments"
     fi
     local -r deb_package="$(pwd)/${1}"
     local -r dest_dir="$(pwd)/${2}"
 
-    local -r extract_dir=$(mktemp -d)
-    cd ${extract_dir}
-    run_and_log_cmd "ar -xv ${deb_package}"
-    mkdir -p files
+    (   
+        # Mender postinstall script uses this variable.
+        export DEVICE_TYPE="${MENDER_DEVICE_TYPE}"
 
-    local -r data_archive=$(find . -maxdepth 1 -type f -regex '.*/data\.tar\.\(zst\|xz\)'  -printf '%P\n' -quit)
-    run_and_log_cmd "sudo tar xf ${data_archive} -C files"
+        run_with_chroot_setup "$dest_dir" deb_install_package_in_chroot "$@"
+    )
+}
 
-    cd - > /dev/null 2>&1
+function deb_install_package_in_chroot() {
+    if [[ $# -ne 2 ]]; then
+        log_fatal "deb_install_package() requires 2 arguments"
+    fi
+    local -r deb_package="$(pwd)/${1}"
+    local -r dest_dir="$(pwd)/${2}"
 
-    run_and_log_cmd "sudo rsync --archive --keep-dirlinks --verbose ${extract_dir}/files/ ${dest_dir}"
+    # Copy package to somewhere reachable from within the chroot.
+    cp "$deb_package" "$dest_dir/tmp/"
+
+    local ret=0
+    run_in_chroot_and_log_cmd_noexit "$dest_dir" "env \
+        DEBIAN_FRONTEND=noninteractive \
+        DEVICE_TYPE=${MENDER_DEVICE_TYPE} \
+        dpkg -i /tmp/$(basename $deb_package)" || ret=$?
```

Comment:
S: `s/-i/--install`

Reply:
Q: Why not install with `apt` straight away? :thinking:

Reply:
AFAIK you cannot install a package file with apt, only a repository package. I did consider doing exactly that in the case when we are requesting the latest package, but then the two code paths would be different, so I chose to keep the dpkg command in both cases.

Reply:
You don't have to do this obviously, but we could consider doing something similar to this: https://github.com/mendersoftware/mender-dist-packages/blob/4cd13030f105bb6f922d57877e468ce174f8fdb7/tests/test_install_mender_sh.py#L99

Reply:
Not sure I see the gain though.. It looks even more complicated than [what we're already doing](https://github.com/mendersoftware/mender-convert/blob/master/modules/deb.sh#L29).